src/hooks/useAutoFocus.ts

```diff
@@ -0,0 +1,24 @@
+import { MutableRefObject, Ref, useEffect, useRef } from 'react';
+
+export const useAutoFocus = <T extends Partial<{ focus: () => any }>>(autoFocus?: boolean, ref?: Ref<T>) => {
```

Comment:
I don't know whether we will need it in the end.
Might be useful if we want to persist autoFocus between re-renders, but not sure.